internal/keysharecore/usersecrets.go

```diff
@@ -31,69 +30,83 @@ var (
 	ErrNoSuchKey              = errors.New("Key identifier unknown")
 )
 
-func (s *unencryptedUserSecrets) pin() [64]byte {
-	var result [64]byte
-	copy(result[:], s[0:64])
-	return result
-}
-
-func (s *unencryptedUserSecrets) setPin(pw [64]byte) {
-	copy(s[0:64], pw[:])
-}
-
-func (s *unencryptedUserSecrets) keyshareSecret() *big.Int {
-	result := new(big.Int)
-	return result.SetBytes(s[64:128])
+func (s *unencryptedUserSecrets) setPin(pin string) error {
+	if len(pin) > 64 {
+		// padBytes also checks the length, but we want to return a specific error in this case
+		return ErrPinTooLong
+	}
+	paddedPin, err := padBytes([]byte(pin), 64)
+	if err != nil {
+		return err
+	}
+	s.Pin = paddedPin
+	return nil
 }
 
 func (s *unencryptedUserSecrets) setKeyshareSecret(val *big.Int) error {
 	if val.Sign() == -1 {
 		return ErrKeyshareSecretNegative
 	}
 
-	data := val.Bytes()
-	if len(data) > 64 {
+	if val.BitLen()/8 > 64 {
```

Comment:
are you sure that this is what you want here? `.Bytes()` really got you the amount of bytes needed to fit all bits. `.BitLen()/8` might return 1 byte too little.

Mini example:
https://go.dev/play/p/IFRJ3T995v0

Reply:
Well spotted, thanks!